scvi/model/_condscvi.py

```diff
@@ -128,34 +128,79 @@ def get_vamp_prior(
         )
         key = labels_state_registry.original_key
         mapping = labels_state_registry.categorical_mapping
-        for ct in range(self.summary_stats.n_labels):
-            # pick p cells
-            local_indices = np.random.choice(
-                np.where(adata.obs[key] == mapping[ct])[0], p
+
+        scdl = self._make_data_loader(adata=adata, batch_size=p)
+
+        mean = []
```

Comment:
add a comment line to explain what's happening here (extracting latent, but getting vars in addition to just the mean)